both/_round.js

```diff
@@ -0,0 +1,175 @@
+
+// Add functions to the Mongo object, using transform (see http://docs.meteor.com/#/full/mongo_collection)
+Round = function(doc) {
+  _.extend(this, doc);
+};
+_.extend(Round.prototype, {
+
+  format: function() {
+    return wca.formatByCode[this.formatCode];
+  },
+  properties: function() {
+    return wca.roundByCode[this.roundCode];
+  },
+  eventName: function() {
+    return this.eventCode ? wca.eventByCode[this.eventCode].name : '';
```

Comment:
Yeah, but I would use a regular exception if barfing is needed.

Anyway, I put in the asserts. Once we break out the ScheduleItem class, eventCode should become mandatory, and this problem disappears in a puff of good design.